solc/CommandLineInterface.cpp

```diff
@@ -1757,13 +822,16 @@ void CommandLineInterface::handleAst()
 
 bool CommandLineInterface::actOnInput()
 {
-	if (m_args.count(g_argStandardJSON) || m_onlyAssemble)
+	if (m_options.inputMode == InputMode::StandardJson || m_options.inputMode == InputMode::Assembler)
 		// Already done in "processInput" phase.
 		return true;
-	else if (m_onlyLink)
+	else if (m_options.inputMode == InputMode::Linker)
```

Comment:
To avoid the diff being all red :)

I actually tried switch at first but back then I had more stuff between the `if`s. Now it's cleaner but I'd still like to keep it at `if`s in this PR. I'll change this in a separate PR.